src/fparser/two/utils.py

```diff
@@ -272,23 +272,52 @@ class Base(ComparableMixin):
                     is either str or FortranReaderBase.
       self.item   - Line instance (holds label) or None.
 
+    :param type cls: the class of object to create.
+    :param string: (source of) Fortran string to parse.
+    :type string: `str` or \
+                  :py:class:`fparser.common.readfortran.FortranReaderBase`
+    :param parent_cls: the parent class of this object.
+    :type parent_cls: `type`
+    :param parent: the parent of this node in the parse tree.
+    :type parent: :py:class:`fparser.two.utils.Base` or NoneType
+
     '''
     # This dict of subclasses is populated dynamically by code at the end
     # of this module. That code uses the entries in the
     # 'subclass_names' list belonging to each class defined in this module.
     subclasses = {}
 
+    def __init__(self, string, parent_cls=None, parent=None):
+        # pylint:disable=unused-argument
+        # During normal construction of the parse tree the parent information
+        # is set-up in the `init` method. However, it may also be provided
+        # to this constructor.
+        self.parent = parent
```

Comment:
This would obviously become `parent = None` if the argument is passed. An alternative would be to have `self._parent=None` and have a parent property to hide the implementation. I'm happy for you to choose which you prefer.

Reply:
I've gone for `self.parent = None`. I don't think hiding it behind a setter/getter buys us anything at the minute (and we can always change later without changing the interface).